starlite/datastructures/state.py

```diff
@@ -12,11 +25,135 @@ class State(MutableMapping[str, Any]):
 
     _state: Dict[str, Any]
 
-    def __init__(self, state: Optional[Dict[str, Any]] = None) -> None:
-        """Initialize `State` with an optional value.
+    def __init__(self, state: Union["ImmutableState", Dict[str, Any], Iterable[Tuple[str, Any]]]) -> None:
+        """Initialize an `ImmutableState` instance.
+
+        Args:
+             state: An object to initialize the state from. Can be a dict, an instance of 'ImmutableState', or a tuple of key value paris.
+        """
+
+        if isinstance(state, ImmutableState):
+            state = state._state
+
+        if not isinstance(state, dict) and isinstance(state, Iterable):
```

Comment:
A deep copy should always be created here, regardless unless `state` is an `ImmutableState` itself. Otherwise you can end up with weird side effects and the `ImmutableState` isn't really immutable.

Reply:
Why deep?

Reply:
Because that would properly follow the semantics. An object that is "frozen" should **never** change. Making a shallow copy would allow situations where it could change.